product/util.py

```diff
@@ -344,9 +345,9 @@ def _effective_images(
 
 def _enumerate_effective_images(
     component_descriptor: ComponentDescriptor,
-    image_reference_filter=lambda _: True,
```

Comment:
please revert this; pass-in a callable. passing an iterable of callables does not add any value, and actually makes the interface less flexible, because it hard-codes that assumption that all filters be ANDed.